connect/api/src/main/java/org/apache/kafka/connect/connector/ConnectRecord.java

```diff
@@ -29,7 +29,7 @@
  * </p>
  */
 @InterfaceStability.Unstable
-public abstract class ConnectRecord {
+public abstract class ConnectRecord<R extends ConnectRecord<R>> {
```

Comment:
According to https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_API_Classes, it is binary compatible.